src/custom_types.rs

```diff
@@ -0,0 +1,77 @@
+use lsp_data::Range;
```

Comment:
Since the language server types crate seemed to only be for types that are part of the spec, this seemed to be the best place for these structs. Are there any concerns with this setup?